### PR TITLE
move test/e2e/framework/lifecycle/ test/framework/cloud/gcp

### PR DESCRIPTION
**Pull request by tanjunchen**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Ref #86052

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

File: test/e2e/lifecycle/framework.go [hunk 1]
```diff
@@ -16,9 +16,102 @@ limitations under the License.
...
+	"k8s.io/apimachinery/pkg/version"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
 
 // SIGDescribe annotates the test with the SIG label.
 func SIGDescribe(text string, body func()) bool {
 	return ginkgo.Describe("[sig-cluster-lifecycle] "+text, body)
 }
+
```
Comment (oomichi) [inline test/e2e/lifecycle/framework.go]: Why do you move these functions into `test/e2e/lifcycle/`?
These functions are used in ` test/e2e/cloud/gcp` only, so these functions should be moved to ` test/e2e/cloud/gcp` I think according to https://github.com/kubernetes/kubernetes/issues/86052

**Review by oomichi — CHANGES_REQUESTED** [review]

File: test/e2e/lifecycle/framework.go (hunk 1 above)
Comment (tanjunchen) [reply test/e2e/lifecycle/framework.go]: Sure

**Comment by tanjunchen** [comment]
ping @oomichi  @SataQiu  done

**Review by SataQiu — COMMENTED** [review]
Thanks @tanjunchen 
/lgtm
/approve

**Pull request merged**